bundles/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/handler/ADBridgeHandler.java

```diff
@@ -0,0 +1,372 @@
+     *
+     * @param msg string containing incoming message payload
+     * @throws MessageParseException
+     */
+    private void parseVersionMessage(String msg) throws MessageParseException {
+        VersionMessage verm;
+
+        try {
+            verm = new VersionMessage(msg);
+        } catch (IllegalArgumentException e) {
+            throw new MessageParseException(e.getMessage());
+        }
+
+        logger.trace("Processing version message sn:{} ver:{} cap:{}", verm.serial, verm.version, verm.capabilities);
+        Map<String, String> properties = editProperties();
+        properties.put(PROPERTY_SERIALNUM, verm.serial);
+        properties.put(PROPERTY_VERSION, verm.version);
+        properties.put(PROPERTY_CAPABILITIES, verm.capabilities);
+        updateProperties(properties);
+    }
+
+    /**
+     * Notify appropriate child thing handlers of an AD message by calling their handleUpdate() methods.
+     *
+     * @param msg message to forward to child handler(s)
+     */
+    private void notifyChildHandlers(ADMessage msg) {
+        for (Thing thing : getThing().getThings()) {
+            ADThingHandler handler = (ADThingHandler) thing.getHandler();
+            //@formatter:off
+            if (handler != null && ((handler instanceof ZoneHandler && msg instanceof EXPMessage) ||
+                                    (handler instanceof RFZoneHandler && msg instanceof RFXMessage) ||
+                                    (handler instanceof KeypadHandler && msg instanceof KeypadMessage) ||
+                                    (handler instanceof LRRHandler && msg instanceof LRRMessage))) {
+                handler.handleUpdate(msg);
+            }
+            //@formatter:on
+        }
+    }
+
+    /**
+     * Notify child thing handlers that the alarm panel is in the ready state. Since there is no way to poll, all
+     * contact channels are initialized into the UNDEF state. This method is called when there is reason to assume that
+     * there are no faulted zones, because the alarm panel is in state READY. Zone handlers that have not yet received
+     * updates can then set their contact states to CLOSED. Only executes the first time panel is ready after bridge
+     * connect/reconnect. Currently only notifies ZoneHandler and RFZoneHandler things.
+     */
+    private void notifyChildHandlersPanelReady() {
+        if (!panelReadyReceived) {
+            panelReadyReceived = true;
+            logger.trace("Notifying child handlers that panel is in ready state");
+
+            // Notify child zone handlers by calling notifyPanelReady() for each
+            for (Thing thing : getThing().getThings()) {
+                ADThingHandler handler = (ADThingHandler) thing.getHandler();
+                if (handler != null && (handler instanceof ZoneHandler || handler instanceof RFZoneHandler)) {
+                    handler.notifyPanelReady();
+                }
+            }
```

Comment:
Technically `instanceof` will only return true for a non-null object, so you don't need to filter them out explicitly.

@J-N-K Also I think this suggested change is less readable than the original. 
In addition, it is less performant and includes the overhead of generating anonymous lambda classes in the bytecode.

Lambdas are fun and useful, but you shouldn't treat them as the only trick in your toolbox otherwise you will try to use them for everything.

Reply:
I think readability is probably a bit personal preference. But I‘m ok to leave it as-is.